File: warehouse/src/main/java/sep3/warehouse/entities/ProductVariant.java
package sep3.warehouse.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;


@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Table(name = "product_variant")
public class ProductVariant {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String size;

    @Column(nullable = false)
    private String material;

    @Column(nullable = false)
    private int stock;

    @ManyToOne
    @JoinColumn(name = "product_id")
    @JsonIgnore
    private Product product;

    @ManyToOne
    @JoinColumn(name = "Archive_Status", referencedColumnName = "id")
    private ArchiveStatus archiveStatus;
}

File: web/Model/ProductVariant.cs
namespace web.Model;

public class ProductVariant
{
    public int Id { get; set; }
    
    public string Size { get; set; }
    
    public string Material { get; set; }
    
    public int Stock { get; set; }

    public long ArchiveStatusId { get; set; }
}